src/lib/db/group-store.ts

```diff
@@ -81,13 +80,23 @@ export default class GroupStore implements IGroupStore {
         return groups.map(rowToGroup);
     }
 
-    async update(group: IGroupModel): Promise<IGroup> {
-        const rows = await this.db(T.GROUPS)
-            .where({ id: group.id })
-            .update(groupToRow(group))
-            .returning(GROUP_COLUMNS);
-
-        return rowToGroup(rows[0]);
+    async update(group: ICreateGroupModel): Promise<IGroup> {
+        try {
+            const rows = await this.db(T.GROUPS)
+                .where({ id: group.id })
+                .update(groupToRow(group))
+                .returning(GROUP_COLUMNS);
+
+            return rowToGroup(rows[0]);
+        } catch (error) {
```

Comment:
If the role doesn't exist at all, it should fail, which I think this now does, thanks to this PR. If the role is not a root role then this will succeed but effectively be a no-op. Probably needs some love but I think this PR takes us forward